lib/mapping/MapFormatH3M.cpp

```diff
@@ -925,8 +925,7 @@ void CMapLoaderH3M::loadArtifactsOfHero(CGHeroInstance * hero)
 		logGlobal->debug("Hero %d at %s has set artifacts twice (in map properties and on adventure map instance). Using the latter set...", hero->getHeroTypeID().getNum(), hero->anchorPos().toString());
 
 		hero->artifactsInBackpack.clear();
-		while(!hero->artifactsWorn.empty())
-			hero->removeArtifact(hero->artifactsWorn.begin()->first);
```

Comment:
This is wrong.

Heroes can have their artifacts configured in two places:
- globally in scenario options
- in hero instance on map

If hero has artifacts configured in both places, game should only use 2nd set, and discard 1st one